NoClassDefFoundError BooleanUtils

**Describe the bug**
NoClassDefFoundError when trying to push a docker image to artifactory with rtDocker.push

```
docker exec .... java -Djava.io.tmpdir=/home/jenkins/workspace/XXX@tmp/artifactory/javatmpdir -cp /home/jenkins/workspace/XXX@tmp/artifactory/cache/artifactory-plugin/3.14.2/* org.jfrog.build.extractor.docker.extractor.DockerPush
Dec 13, 2021 6:36:06 PM org.jfrog.build.extractor.packageManager.PackageManagerLogger info
INFO: Pushing image: XXX
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/commons/lang/BooleanUtils
	at com.github.dockerjava.core.DefaultDockerClientConfig$Builder.build(DefaultDockerClientConfig.java:442)
	at org.jfrog.build.extractor.docker.DockerJavaWrapper.getDockerClient(DockerJavaWrapper.java:76)
  	at org.jfrog.build.extractor.docker.DockerJavaWrapper.pushImage(DockerJavaWrapper.java:41)
  	at org.jfrog.build.extractor.docker.extractor.DockerPush.execute(DockerPush.java:84)
  	at org.jfrog.build.extractor.packageManager.PackageManagerExtractor.executeAndSaveBuildInfo(PackageManagerExtractor.java:34)
  	at org.jfrog.build.extractor.docker.extractor.DockerPush.main(DockerPush.java:69)
  Caused by: java.lang.ClassNotFoundException: org.apache.commons.lang.BooleanUtils
  	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
  	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
  	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
  	... 6 more
```

**To Reproduce**
Got this error after updating from Jenkins 2.303.2 and Artifactory plugin 3.13.2 to Jenkins 2.319.1 and Artifactory plugin 3.14.2

**Expected behavior**
rtDocker.push should be able to execute

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Versions**
- Jenkins Artifactory plugin version: 3.14.2
- Jenkins operating system: 2.319.1
- Artifactory Version: 7.27.6

**Additional context**
Add any other context about the problem here.

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -82,6 +82,16 @@ jobs:
           echo "MAVEN_HOME=$(which mvn)" >> $GITHUB_ENV
           echo "GRADLE_HOME=$(brew info gradle | grep /usr/local/Cellar/gradle | awk '{print $1}')" >> $GITHUB_ENV
 
+      # Install docker-java locally
+      - name: Install docker-java
+        run: |
+          git clone https://github.com/docker-java/docker-java.git
+          cd docker-java
+          git checkout 8e77a694
+          ./mvnw versions:set -DnewVersion=3.2.x-8e77a694
+          ./mvnw install -DskipTests
+          cd ..
+
       - name: Run tests
         env:
           JENKINS_PLATFORM_URL: ${{ secrets.JENKINS_PLATFORM_URL }}

--- a/pom.xml
+++ b/pom.xml
@@ -382,6 +382,18 @@
             <groupId>org.jfrog.buildinfo</groupId>
             <artifactId>build-info-extractor-docker</artifactId>
             <version>${buildinfo.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>com.github.docker-java</groupId>
+                    <artifactId>docker-java</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <!-- Warning - this is a temporary workaround supposed to be removed after the next release of docker-java -->
+            <groupId>com.github.docker-java</groupId>
+            <artifactId>docker-java</artifactId>
+            <version>3.2.x-8e77a694</version>
         </dependency>
         <dependency>
             <groupId>org.jfrog.buildinfo</groupId>

--- a/release/pipelines.release.yml
+++ b/release/pipelines.release.yml
@@ -46,6 +46,14 @@ pipelines:
             - test -n "$NEXT_VERSION" -a "$NEXT_VERSION" != "0.0.0"
             - test -n "$NEXT_DEVELOPMENT_VERSION" -a "$NEXT_DEVELOPMENT_VERSION" != "0.0.0"
 
+            # Install docker-java
+            - git clone https://github.com/docker-java/docker-java.git
+            - cd docker-java
+            - git checkout 8e77a694
+            - ./mvnw versions:set -DnewVersion=3.2.x-8e77a694
+            - ./mvnw install -DskipTests
+            - cd ..
+
             # Configure JFrog CLI
             - curl -fL https://install-cli.jfrog.io | sh
             - jf c rm --quiet

--- a/release/pipelines.snapshot.yml
+++ b/release/pipelines.snapshot.yml
@@ -31,6 +31,14 @@ pipelines:
             - export JFROG_CLI_BUILD_NUMBER=$run_number
             - export JFROG_CLI_BUILD_PROJECT=ecosys
 
+            # Install docker-java
+            - git clone https://github.com/docker-java/docker-java.git
+            - cd docker-java
+            - git checkout 8e77a694
+            - ./mvnw versions:set -DnewVersion=3.2.x-8e77a694
+            - ./mvnw install -DskipTests
+            - cd ..
+
             # Configure JFrog CLI
             - curl -fL https://install-cli.jfrog.io | sh
             - jf c rm --quiet

